src/components/AppController.tsx

```diff
@@ -78,40 +78,73 @@ interface AppControllerProps {
  * @return
  */
 const AppController = ({children}: AppControllerProps) => {
-    const {filePath, isPrettified, logEventNum} = useContext(UrlContext);
-
     // States
-    const setLogEventNum = useContextStore((state) => state.setLogEventNum);
-    const logFileManagerProxy = useLogFileManagerStore((state) => state.logFileManagerProxy);
+    const beginLineNumToLogEventNum = useViewStore((state) => state.beginLineNumToLogEventNum);
+
+    const fileSrc = useLogFileStore((state) => state.fileSrc);
     const loadFile = useLogFileStore((state) => state.loadFile);
+    const setFileSrc = useLogFileStore((state) => state.setFileSrc);
+
+    const isPrettified = useViewStore((state) => state.isPrettified);
+    const updateIsPrettified = useViewStore((state) => state.updateIsPrettified);
+
+    const {logFileManagerProxy} = useLogFileManagerStore.getState();
     const numEvents = useLogFileStore((state) => state.numEvents);
-    const beginLineNumToLogEventNum = useViewStore((state) => state.beginLineNumToLogEventNum);
-    const setIsPrettified = useViewStore((state) => state.updateIsPrettified);
+
+    const logEventNum = useViewStore((state) => state.logEventNum);
+    const setLogEventNum = useViewStore((state) => state.setLogEventNum);
+
     const updatePageData = useViewStore((state) => state.updatePageData);
+
     const setUiState = useUiStore((state) => state.setUiState);
 
     // Refs
-    const isPrettifiedRef = useRef<boolean>(isPrettified ?? false);
+    const isPrettifiedRef = useRef<boolean>(isPrettified);
     const logEventNumRef = useRef(logEventNum);
```

Comment:
fwiw, those refs are no longer needed. thanks to Zustand, in the useEffect hooks we can read the latest value of some Zustand state variable using
```
const {isPrettified} = useViewStore.getState();
```
for example